crates/miden-lib/asm/kernels/transaction/lib/epilogue.masm

```diff
@@ -286,7 +307,18 @@ export.finalize_transaction
 
     # compute the fee the tx needs to pay
     # ignored for now
-    exec.compute_fee drop
+    exec.compute_fee
+    # => [fee_amount]
+
+    exec.build_native_fee_asset
+    # => [FEE_ASSET]
+
+    # remove the fee from the native account's vault
+    exec.account::remove_asset_from_vault
+    # => [FEE_ASSET]
```

Comment:
This is probably fine for now, but I wonder if these 3 calls should be encapsulated into a single procedure - something like `pay_fee_from_account`. Mostly thinking about the future where fee payments could come from other places - though, this is probably premature for now.